servers/visual/visual_server_canvas.cpp

```diff
@@ -754,12 +753,7 @@ void VisualServerCanvas::canvas_item_add_polygon(RID p_item, const Vector<Point2
 	ERR_FAIL_COND(uv_size != 0 && (uv_size != pointcount));
 #endif
 	Vector<int> indices = Geometry::triangulate_polygon(p_points);
-
-	if (indices.empty()) {
-
-		ERR_EXPLAIN("Bad Polygon!");
-		ERR_FAIL();
-	}
+	ERR_FAIL_COND_MSG(indices.empty(), "Bad Polygon!");
```

Comment:
Let's take the opportunity to make it more explicit, e.g. "Invalid polygon data, triangulation failed.".